### ISSUE TITLE
tailwindFunctions option with twin.macro / tailwind-styled-components

### ISSUE THREAD

**Issue by AleksiVirkkala** [tailwindlabs/prettier-plugin-tailwindcss#167]
**What version of `prettier-plugin-tailwindcss` are you using?**

v0.3.0

**What version of Tailwind CSS are you using?**

v3.3.2

**What version of Node.js are you using?**

v18.16.0

**What package manager are you using?**

pnpm

**What operating system are you using?**

macOS

**Reproduction**

`prettier.config.js`:

```js
module.exports = {
  plugins: [require('prettier-plugin-tailwindcss')],
  tailwindFunctions: ['tw', 'tw*', 'tw.div', 'div'] // none of these work
};
```
`Example.tsx`:
```ts
import tw from 'tailwind-styled-components';

// Correct syntax
tw.div`px-4 h-max`; // doesn't work
tw['div']`px-4 h-max`; // doesn't work

// Invalid syntax
tw`px-4 h-max`; // works
div`px-4 h-max`; // works

```
**Describe your issue**

This has been discussed e.g. here: [11216](https://github.com/tailwindlabs/tailwindcss/discussions/11216)

At minimum, I'd like to be able to in some way be able to configure this plugin to work with styled component like syntax, e.g.:

```ts
tw.div`px-4 h-max`
```

However at the moment this does not work even if I tried to hardcode every element to `tailwindFunctions` option in `prettier.config.js`.

As an optimal solution I think it would be great to be able to pass regex for this option. TailwindCSS VS-Code plugin supports this feature through experimental editor option which makes it possible to configure intellisense ([more here](https://github.com/MathiasGilson/tailwind-styled-component#optional-configure-intellisense-autocomplete-on-vscode)).

`.vscode/settings.json`:

```json5
"tailwindCSS.experimental.classRegex": [
  "tw`([^`]*)", // tw`...`
  "tw\\.[^`]+`([^`]*)`", // tw.xxx<xxx>`...`
  "tw\\(.*?\\).*?`([^`]*)" // tw(Component)<xxx>`...`
]
```

**Comment by thecrypticace** [comment]
Hey! I've merged in a change that fixes this in #169. I _think_ we want `tw`, `tw.div`, `tw["div"]`, `tw.foo.bar`, etc… to just work when you specify `tw`. But not yet 100% certain so will hold off on tagging a release until that decision has been made (should be this week). If it does change I'll open a second PR with the additional changes and mention it here as well.

In the meantime you can, however, give this change a test using our insiders build:
```
npm install prettier-plugin-tailwindcss@insiders
```

**Comment by imranbarbhuiya** [comment]
sorry for commenting on an old issue but the above pr only fixed 
```js
tw.div`px-4 h-max`;
```
 issue but we still have `tw.style('px-4 h-max)` with function call issues. Should I create a new issue for that?

**Comment by thecrypticace** [comment]
@imranbarbhuiya normally yeah but I went ahead and fixed this. Released as v0.5.5.

**Comment by imranbarbhuiya** [comment]
Thanks ❤️
